### PR TITLE
[master] Fix the bug caused by oversized packets

### PR DESCRIPTION
**Pull request by Tim-Zhang**
Fixes: https://github.com/containerd/ttrpc-rust/issues/198

### CODE REVIEW

File: src/asynchronous/server.rs [hunk 1]
```diff
@@ -413,21 +414,26 @@ impl HandlerContext {
...
         let stream_id = msg.header.stream_id;
 
-        if (stream_id % 2) != 1 {
-            Self::respond_with_status(
-                self.tx.clone(),
-                stream_id,
-                get_status(Code::INVALID_ARGUMENT, "stream id must be odd"),
-            )
-            .await;
+        if let Err(e) = msg.check() {
```
Comment (lifupan) [inline src/asynchronous/server.rs:417]: Here's check is also redundant?

File: src/asynchronous/stream.rs [hunk 2]
```diff
@@ -418,6 +418,9 @@ impl StreamSender {
             header,
             payload: buf,
         };
+
```
Comment (lifupan) [inline src/asynchronous/stream.rs:421]: Here's check is also redundant?

File: src/asynchronous/stream.rs [hunk 3]
```diff
@@ -447,6 +450,9 @@ impl StreamReceiver {
             return Err(Error::RemoteClosed);
         }
         let msg = _recv(&mut self.rx).await?;
+
+        msg.check()?;
```
Comment (lifupan) [inline src/asynchronous/stream.rs:454]: Here's check is also redundant?

File: src/asynchronous/stream.rs (hunk 2 above)
Comment (Tim-Zhang) [reply src/asynchronous/stream.rs:421]: We need check it before stream's sending

File: src/asynchronous/client.rs [hunk 4]
```diff
@@ -97,6 +97,9 @@ impl Client {
         };
 
         let msg = result?;
+        // Check received server response
+        msg.check()?;
```
[the remark this answers is not in the document]
Comment (Tim-Zhang) [reply src/asynchronous/client.rs]: removed

File: src/asynchronous/stream.rs (hunk 3 above)
Comment (Tim-Zhang) [reply src/asynchronous/stream.rs]: removed

**Review by lifupan — APPROVED** [review]

**Review by teawater — APPROVED** [review]

**Pull request merged**